test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,237 @@
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed)
+                   | seqan3::views::persist
+                   | seqan3::views::to_rank
+                   | std::views::transform([] (uint8_t const r) { return std::min<uint8_t>(r, 253); })
+                   | seqan3::views::rank_to<alphabet_t>
+                   | seqan3::views::to<inner_rng_t>;
+        }
+        else
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed) | seqan3::views::to<inner_rng_t>;
+        }
+    }
+
+    rng_t make_collection_seqan3()
+    {
+        rng_t result;
+        for (size_t i = 0; i < collection_size; ++i)
+            result.emplace_back(make_single_seqan3(i));
+
+        return result;
+    }
+
+    rng_t make_sequence_seqan3()
+    {
+        if constexpr(seqan3::dimension_v<rng_t> == 1)
+            return make_single_seqan3();
+        else
+            return make_collection_seqan3();
+    }
+
+#if SEQAN3_HAS_SEQAN2
+    // Map the SeqAn3 alphabet to its SeqAn2 equivalent.
+    using seqan2_alphabet_t = std::conditional_t<std::same_as<alphabet_t, seqan3::dna4>, seqan::Dna,
+                                std::conditional_t<std::same_as<alphabet_t, seqan3::aa27>, seqan::AminoAcid, char>>;
+    // SeqAn2: `seqan::String` and `seqan::StringSet`, SeqAn3: `std::vector` and `std::vector<std::vector>`.
+    using seqan2_rng_t = std::conditional_t<seqan3::dimension_v<rng_t> == 1,
+                                            seqan::String<seqan2_alphabet_t>,
+                                            seqan::StringSet<seqan::String<seqan2_alphabet_t>>>;
+    using index_cfg = seqan::FastFMIndexConfig<void, uint32_t>;
+    using fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::FMIndex<void, index_cfg>>;
+    using bi_fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>>;
+
+    template <typename to_t, typename from_t, template <typename> typename range_t>
+        requires (seqan3::alphabet<from_t> && std::ranges::forward_range<range_t<from_t>>)
+    seqan::String<to_t> convert_single_range(range_t<from_t> const & original_range)
+    {
+        seqan::String<to_t> result_range;
+        seqan::reserve(result_range, std::ranges::size(original_range));
+
+        for (auto const & element : original_range)
+            appendValue(result_range, seqan3::to_char(element));
+
+        return result_range;
+    }
+
+    template <typename to_t, typename from_t, template <typename> typename range_t>
+        requires (seqan3::alphabet<from_t> && std::ranges::forward_range<range_t<from_t>>)
+    seqan::StringSet<seqan::String<to_t>> convert_range_of_range(std::vector<range_t<from_t>> const & original_range)
+    {
+        seqan::StringSet<seqan::String<to_t>> result_range;
+
+        for (auto const & single_range : original_range)
+            appendValue(result_range, convert_single_range<to_t>(single_range));
```

Comment:
This is my idea from above:
```suggestion
        {
            seqan::reserve(result_range, std::ranges::size(original_range));
            std::transform(std::ranges::begin(single_range), std::ranges::end(single_range), 
                           seqan::begin(result_range), seqan3::to_char);
        }
```

Reply:
Ah, but with `result_range` as the actual element inside the string set.